<launch>

  <!-- Generate target motion -->
  <!-- ====================== -->

  <!-- Figure of eight motion -->
  <node pkg="ik_ros_examples" name="figure_eight_motion" type="figure_eight_node.py">
    <param name="start_on_init" type="bool" value="false"/>
  </node>

  <!-- Connect figure of eight motion to world -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="motion_base_in_world"
	args="-0.5 0 0.65 0 1.57 3.141 rpbi/world figure_eight_base"/>

  <!-- IK -->
  <!-- == -->

  <!-- Setup IK -->
  <node pkg="ik_ros" name="ik_setup" type="ik_setup_node.py" args="pybullet" output="screen">
    <rosparam param="config" file="$(find ik_ros_examples)/configs/kuka_lwr_pybullet_example/ik_setup_config.yaml"/>
    <param name="start_on_init" type="bool" value="false"/>
    <remap from="ik/setup/pybullet/currentPosition" to="rpbi/kuka_lwr/joint_states"/>
    <remap from="ik" to="rpbi/kuka_lwr/ik"/>
  </node>

  <!-- NOTE: no need for solver node for 'pybullet' IK interface. See ik_ros/src/ik_ros/pybullet/README.md -->

  <!-- Simulation -->
  <!-- ========== -->

  <!-- ROS-Pybullet interface node simulator -->
  <node pkg="ros_pybullet_interface" name="ros_pybullet_interface_node" type="ros_pybullet_interface_node.py">
    <rosparam param="config" file="$(find ik_ros_examples)/configs/kuka_lwr_pybullet_example/config.yaml"/>
  </node>

  <!-- Visualization -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find ik_ros_examples)/rviz/kuka_lwr_example.rviz" />

  <!-- Run example -->
  <!-- =========== -->

  <node pkg="ik_ros_examples" name="start" type="run_example.py" args="pybullet">
    <param name="robot_name" type="str" value="kuka_lwr"/>
    <param name="eff_name" type="str" value="lwr_arm_7_link"/>
  </node>

</launch>
